[gmock] Support Production builds

JonWBedard opened the pull request:
#### 1d6b1cdf6655219d0c0ff66b474e88031ec19827
<pre>
[gmock] Support Production builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=289707">https://bugs.webkit.org/show_bug.cgi?id=289707</a>
<a href="https://rdar.apple.com/146955834">rdar://146955834</a>

Reviewed by Elliott Williams.

* .submitproject-tools: Add gmock.
* Source/ThirdParty/gmock/Configurations/Base.xcconfig: Correct CommonBase include path.
* Source/ThirdParty/gmock/Configurations/gmock.xcconfig: Install gmock.
* Source/ThirdParty/gmock/gmock.xcodeproj/project.pbxproj: Include all headers in
installhdrs, install gmock to SDK.
* WebKit.xcworkspace/contents.xcworkspacedata: Include gmock project.

Canonical link: <a href="https://commits.webkit.org/292120@main">https://commits.webkit.org/292120@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/61b631118d33096c617db85485179d6ca5c1f5a3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95008 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14603 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4460 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100037 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45506 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/97057 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14892 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23025 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72471 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29767 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98009 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11109 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85784 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52798 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10820 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3509 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44845 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/81017 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3603 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102082 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22050 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16093 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/81475 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22297 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81800 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80860 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20206 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25422 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2824 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15292 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22023 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21680 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25155 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23421 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on WebKit.xcworkspace/contents.xcworkspacedata:
```diff
@@ -40,6 +40,9 @@
       location = "group:Source/ThirdParty/gtest/xcode/gtest.xcodeproj">
    </FileRef>
+   <FileRef
+      location = "container:Source/ThirdParty/gmock/gmock.xcodeproj">
```
I don't know the difference between "group:" and "container:" in this format, but it does strike me that this one is different from all the rest. And I suspect we don't actually need gmock in the workspace, because it's already available through nested projects, and Apple's internal build builds individual projects and not schemes in the workspace.

JonWBedard replied:
It's actually not available through nested projects, at the moment. I did this through the UI, I'll change this to match everything else.

emw-apple commented on Source/ThirdParty/gmock/Configurations/Base.xcconfig:
```diff
@@ -21,7 +21,7 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-#include "../../../../Configurations/CommonBase.xcconfig"
+#include "../../../Configurations/CommonBase.xcconfig"
```
Nice 😆

emw-apple commented on Source/ThirdParty/gmock/gmock.xcodeproj/project.pbxproj:
```diff
@@ -57,9 +74,6 @@
 			dstPath = "$(PUBLIC_HEADERS_FOLDER_PATH)/internal";
 			dstSubfolderSpec = 16;
 			files = (
-				440034652AD8C1BC0081CD73 /* gmock-internal-utils.h in Copy Headers Internal */,
-				440034662AD8C1BF0081CD73 /* gmock-port.h in Copy Headers Internal */,
-				440034672AD8C1C10081CD73 /* gmock-pp.h in Copy Headers Internal */,
```
Please go ahead and remove this whole copy files phase, since we're now handling everything through the main Headers phase.

emw-apple commented on Source/ThirdParty/gmock/gmock.xcodeproj/project.pbxproj:
```diff
@@ -70,9 +84,6 @@
 			dstPath = "$(PUBLIC_HEADERS_FOLDER_PATH)/internal/custom";
 			dstSubfolderSpec = 16;
 			files = (
-				440034682AD8C1C50081CD73 /* gmock-generated-actions.h in Copy Headers Internal Custom */,
-				440034692AD8C1C80081CD73 /* gmock-matchers.h in Copy Headers Internal Custom */,
-				4400346A2AD8C1CA0081CD73 /* gmock-port.h in Copy Headers Internal Custom */,
```
Ditto

JonWBedard committed:
[gmock] Support Production builds
https://bugs.webkit.org/show_bug.cgi?id=289707
rdar://146955834

Reviewed by Elliott Williams.

* .submitproject-tools: Add gmock.
* Source/ThirdParty/gmock/Configurations/Base.xcconfig: Correct CommonBase include path.
* Source/ThirdParty/gmock/Configurations/gmock.xcconfig: Install gmock.
* Source/ThirdParty/gmock/gmock.xcodeproj/project.pbxproj: Include all headers in
installhdrs, install gmock to SDK.
* WebKit.xcworkspace/contents.xcworkspacedata: Include gmock project.

Canonical link: https://commits.webkit.org/292120@main
